extensions/amazon-lambda-http/deployment/src/main/resources/http/sam.jvm.yaml

```diff
@@ -17,6 +17,10 @@
         MemorySize: 512
         Policies: AWSLambdaBasicExecutionRole
         Timeout: 15
+        Environment:
+          Variables:
+            QUARKUS_AWS_LAMBDA_FORCE_USER: false
```

Comment:
why do we need 2 vars ? I mean if we jave the USER_NAME the let's use it, if not let's ignore that. The QUARKUS_AWS_LAMBDA_FORCE_USER looks redundant to me, what do you think ?

Reply:
I agree.  One var is enough.

Reply:
Correct me if I'm wrong (and I'm not an AWS SAM expert), but I thought that in order to be able to set an environment variable to a particular value in SAM, it [needs to be declared in the template](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-using-invoke.html#serverless-sam-cli-using-invoke-environment-file) with a value:

> You do this to provide a JSON file that contains values to override the environment variables that are **already defined** in your function template.

(emphasis mine)

If you want to then change that value at runtime, you either need to change the template or provide a different parameters file.

This means that parameters you *may* want to use must always be defined in the template with a value. Which means that if you want to do this (again, from memory, may not be quite right):

`sam start-api --template path/to/template.yaml`

(ie without 'hacking' the JWT token)

and you also want to do this:

`sam start-api --template path/to/template.yaml --env-vars runAsUserAlbus.json`

and also this:

`sam start-api --template path/to/template.yaml --env-vars runAsUserHarry.json`

then template.yaml always needs to have a value for the variables you might need. Which means you need a variable to turn the feature on/off and a variable to carry the username, because the username variable on its own will always have a username.

But like I said, that might be wrong and might have misunderstood the documentation. Don't hesitate to correct me!

Reply:
Just seems weird to be that you have to enable the config switch to set it.  Would a default value for the variable be better and if that default variable is set ignore the switch?

What if this variable is not in the generated yaml template and instead its just documented?  If you want your lambda to RUN AS a specific principal in SAM LOCAL, set this envronment variable in your template.

Reply:
Yes, you're right, let's remove the configuration from the templates, and make stick it all in the documentation. I'll update the templates. I'll also sort out some tests but that may take a day or two.